MLS.Agent/DotnetKernelJupyterInstaller.cs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.IO;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+using MLS.Agent.Tools;
+
+namespace MLS.Agent
+{
+    public static class DotnetKernelJupyterInstaller
+    {
+        public delegate Task<CommandLineResult> ExecuteCommand(string command, string args);
+
+        private static async Task<int> TryInstallKernel(ExecuteCommand executeCommand, IConsole console)
+        {
+            try
+            {
+                var result = await executeCommand("jupyter", "--paths");
+                if(result.ExitCode ==0)
+                {
+                    var dataDirectories = await GetJupyterDataPaths(result.Output);
+                    Installkernel(dataDirectories, console);
+                    console.Out.WriteLine(".NET kernel installation succeded");
+                    return 0;
+                }
+                else
+                {
+                    throw new KernelInstallationFailureException($"Tried to invoke \"jupyter --paths\" but failed with exception: {string.Join("\n", result.Error)}");
```

Comment:
It's best to avoid using exceptions for control flow. Since this exception will always be caught and result in the behavior inside your catch block, you can simplify this by just putting the console output and result code inside the else block and getting rid of the `try`/`catch` and exception altogether.